### PR TITLE
Add unit test for GetNewRC and GetOldRCs in deployment

### PR DESCRIPTION
**Pull request by janetkuo**
Unit test for `GetNewRC` and `GetOldRCs`. 
cc @nikhiljindal @kargakis @bgrant0607

### CODE REVIEW

**Comment by kargakis** [comment]
Looks good. Add tests also for GetOldRCs.

**Comment by janetkuo** [comment]
Yes I'll add tests for GetOldRCs too.

**Comment by nikhiljindal** [comment]
This LG.
I assume you will add tests for GetOldRCs as well in this PR?

**Comment by janetkuo** [comment]
@nikhiljindal yes I'll add the unit test for GetOldRCs in this PR. Just wanted to get some earlier feedback before doing both.

File: pkg/util/deployment/deployment_test.go [hunk 1]
```diff
@@ -75,3 +79,256 @@ func TestGetReadyPodsCount(t *testing.T) {
...
+					generateRC(generateDeployment("xyz")),
+				},
+			},
+			&newRC,
+		},
+	}
+
+	ns := api.NamespaceDefault
+	for _, test := range tests {
+		c := &simple.Client{
```
Comment (janetkuo) [inline pkg/util/deployment/deployment_test.go]: I can only set one request in simple.Client, but I need GET both replicationcontrollers and pods for GetOldRCs, so I changed the function to only test its main logic.

**Comment by janetkuo** [comment]
@nikhiljindal I have a question about `GetOldRCs`.  In its implementation, we first list all pods with the deployment's selector, and then list all rcs. Then we find the rcs' selector that matches pods (the rcs that manage the pods) and then check if its template is different from the new template; if so, we add it to old rcs. 

If we have a deployment whose selector is `name=nginx`, an rc whose selector is also `name=nginx` (without podtemplatehash label) and its pod whose label is `name=nginx`. If the rc isn't created by the deployment, `GetOldRCs` will still see it as an old RC of this deployment. Is this expected?

File: pkg/util/deployment/deployment_test.go (hunk 1 above)
Comment (kargakis) [inline pkg/util/deployment/deployment_test.go]: You may want to use a fake client and register multiple reactions.

**Comment by nikhiljindal** [comment]
Yes we want to allow deployment to adopt existing RCs.

**Comment by janetkuo** [comment]
@nikhiljindal thanks I'll add test cases for that too.

File: pkg/util/deployment/deployment_test.go (hunk 1 above)
Comment (janetkuo) [inline pkg/util/deployment/deployment_test.go]: Thanks! Code updated.

**Comment by janetkuo** [comment]
PTAL

**Comment by nikhiljindal** [comment]
LGTM, thanks!

**Pull request merged**